clang/test/Sema/builtin-masked.m

```diff
@@ -0,0 +1,15 @@
+// RUN: %clang_cc1 -fsyntax-only -fobjc-arc -verify %s
+
+typedef int v8i __attribute__((ext_vector_type(8)));
+typedef _Bool v8b __attribute__((ext_vector_type(8)));
+
+@interface Obj @end
+
+void objc(v8b mask, __strong Obj * ptr, v8i v) {
+  (void)__builtin_masked_load(mask, ptr); // expected-error {{2nd argument must be a scalar pointer}}
+  (void)__builtin_masked_store(mask, v, ptr); // expected-error {{3rd argument must be a scalar pointer}}
+  (void)__builtin_masked_expand_load(mask, ptr); // expected-error {{2nd argument must be a scalar pointer}}
+  (void)__builtin_masked_compress_store(mask, v, ptr); // expected-error {{3rd argument must be a scalar pointer}}
+  (void)__builtin_masked_gather(mask, v, ptr); // expected-error {{3rd argument must be a scalar pointer}}
+  (void)__builtin_masked_scatter(mask, v, v, ptr); // expected-error {{4th argument must be a scalar pointer}}
```

Comment:
Okay, this isn't quite was I was asking for. The pattern to follow here is `test/Sema/matrix-type-builtins.c`.

You want to declare a bunch of properties that have types that would be useful (or not) for your builtin, like this:

```
@interface Obj

@property int *int_ptr;
@property v8i int_vector;
@property int **int_ptr_ptr;

@end
```

Now if `ptr` is an `Obj*`, `ptr.int_ptr` should produce an `int*`, `ptr.int_vector` should produce a `v8i`, etc. You just want to take one of your basic tests that checks that the builtins work with variables of those types and change it use those expressions instead of variables.